dandiapi/api/views/asset.py

```diff
@@ -194,20 +199,26 @@ def update(self, request, versions__dandiset__pk, versions__version, **kwargs):
                 status=status.HTTP_405_METHOD_NOT_ALLOWED,
             )
 
-        serializer = AssetRequestSerializer(data=request.data)
+        serializer = AssetUpdateRequestSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
 
-        asset_blob = get_object_or_404(AssetBlob, blob_id=serializer.validated_data['blob_id'])
-
-        metadata = serializer.validated_data['metadata']
-        if 'path' not in metadata:
-            return Response('No path specified in metadata', status=404)
-        path = metadata['path']
-        # Strip away any computed fields
-        metadata = Asset.strip_metadata(metadata)
-        asset_metadata, created = AssetMetadata.objects.get_or_create(metadata=metadata)
-        if created:
-            asset_metadata.save()
+        if 'blob_id' in serializer.validated_data:
+            asset_blob = get_object_or_404(AssetBlob, blob_id=serializer.validated_data['blob_id'])
```

Comment:
If we decide to deprecate providing `blob_id` altogether (#392) we might better here test that it is the same asset here as the `old_asset.blob` instead of allowing for creating an asset for another blob here on PUT.  And if mismatch -- error out.

WDYT @satra?

Reply:
PUT on an asset has three possibilities:
1. change path
2. change blob
3. change metadata (other than path or blob)

and in all cases unlink the previous asset and create a new asset. (everything could be seen as metadata update). so we are using PUT instead of DELETE + POST

so i would be hesitant to change what PUT does, unless we enforce those semantics consistently. i.e. 
- PUT does not change asset id and cannot be used to change blob
- replacement blob would require explicit DELETE + POST

Reply:
It seems like this discussion should be part of that proposed new issue as well.